src/type/validate.ts

```diff
@@ -433,6 +433,21 @@ function validateTypeImplementsInterface(
         );
       }
     }
+
+    // Asserts that field is not deprecated unless interface field is
+    if (
+      ifaceField.deprecationReason == null &&
+      typeField.deprecationReason != null
```

Comment:
super nit: flip the order of these two so it reads the same as the spec? "if the type's field is deprecated, then the interface's field must be deprecated"